src/server/scripts/Spells/spell_shaman.cpp

```diff
@@ -456,6 +460,85 @@ class spell_sha_earthen_rage_proc_aura : public AuraScript
     }
 };
 
+// 61882 - Earthquake
+class spell_sha_earthquake : public SpellScript
+{
+    PrepareSpellScript(spell_sha_earthquake);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SHAMAN_AFTERSHOCK, SPELL_SHAMAN_AFTERSHOCK_ENERGIZE });
+    }
+
+    void HandleAfterCast()
+    {
+        int32 energize = *GetSpell()->GetPowerTypeCostAmount(POWER_MAELSTROM);
+
+        if (AuraEffect* aftershock = GetCaster()->GetAuraEffect(SPELL_SHAMAN_AFTERSHOCK, EFFECT_0))
+            if (roll_chance_f(aftershock->GetAmount()))
+                GetCaster()->CastSpell(GetCaster(), SPELL_SHAMAN_AFTERSHOCK_ENERGIZE, CastSpellExtraArgs(energize)
+                    .AddSpellBP0(energize));
+    }
+
+    void Register() override
+    {
+        AfterCast += SpellCastFn(spell_sha_earthquake::HandleAfterCast);
+    }
+};
+
+// 61882 - Earthquake aura
+class spell_sha_earthquake_aura : public AuraScript
+{
+    PrepareAuraScript(spell_sha_earthquake_aura);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SHAMAN_EARTHQUAKE, SPELL_SHAMAN_EARTHQUAKE_TICK });
+    }
+
+    void HandlePeriodic(AuraEffect const* /*aurEff*/)
+    {
+        if (Unit* caster = GetCaster())
+        {
+            std::vector<AreaTrigger*> areaTriggers = caster->GetAreaTriggers(SPELL_SHAMAN_EARTHQUAKE);
+
+            for (AreaTrigger* at : areaTriggers)
+                caster->CastSpell(at->GetPosition(), SPELL_SHAMAN_EARTHQUAKE_TICK, true);
```

Comment:
This would synchronize all ticks on all earthquakes cast by player (not how it behaves on retail)

Reply:
@Robingadko i would propose that you create a areatrigger script for this case and calculate the tick time inside the areatrigger script and then just let the Caster cast SPELL_SHAMAN_EARTHQUAKE_TICK at the AT position